pkg/store/proxy.go

```diff
@@ -198,9 +201,37 @@ func (s ctxRespSender) send(r *storepb.SeriesResponse) {
 	s.ch <- r
 }
 
+func matchesRequiredMatchers(ms []storepb.LabelMatcher, requiredMatcher labels.Labels) (bool, error) {
```

Comment:
This function is almost the same as the function `matchesExternalLabels` below.  What about using that function directly? Maybe we can rename `matchesExternalLabels` to sth else. WDYT?

Reply:
Yeah we can use `matchesExternalLabels` with the following logic
```go
ok, newmatchers, err := matchesExternalLabels(r.Matchers, s.requiredMatcher)
if err != nil {
	return status.Error(codes.InvalidArgument, err.Error())
}
if !ok {
	return nil
}
if len(newmatchers) > len(r.Matchers)-len(s.requiredMatcher) {
	return nil
}
```
WDYT?

Reply:
Hm, I don't think this is the same. This is totally agnostic to external matchers at current state.